block_filtered_course_list.php

```diff
@@ -207,24 +207,21 @@ private function _process_filtered_list() {
         global $PAGE;
         $output = $PAGE->get_renderer('block_filtered_course_list');
 
-        // Parse the textarea settings into an array of arrays.
-        $filterconfigs = array_map(function($line) {
-            return array_map(function($item) {
-                return trim($item);
-            }, explode('|', $line, 2));
-        }, explode("\n", $this->fclconfig->filters));
-
+        // Parse the textarea settings into an array of lines.
+        $filterconfig = new \block_filtered_course_list\local\config_parser($this->fclconfig->filters);
         // Get the arrays of rubrics based on the config lines, filter out failures, and merge them into one array.
         $this->rubrics = array_reduce(
             array_filter(
-                array_map(function($config) {
-                    $classname = "block_filtered_course_list_{$config[0]}_configline";
-                    if (class_exists($classname)) {
-                        $item = new $classname($config, $this->mycourses, $this->fclconfig);
-                        return $item->get_rubrics();
-                    }
-                    return null;
-                }, $filterconfigs), function($item) {
+                array_map(
+                    function($config) {
+                        $classname = "\\block_filtered_course_list\\local\\filter_{$config[0]}\\filter";
```

Comment:
So just to confirm my understanding: This naming works somewhat magically because you are putting this code into the classes folder? I can see the appeal.

Reply:
Well the namespaces are explicitly declared at the top of each class file. And Moodle requires that the directory structure match the namespace structure in order for it to autoload the classes.

Reply:
That is what I had understood, so that seems good to me if you want to start by splitting things out this way.